### PR TITLE
DOC: Removing bad practices from quick start + some PEP8

### PR DESCRIPTION
**Pull request by alexdesiqueira**
Replacing `.shape = ` with `.reshape` + some PEP8 changes.

### CODE REVIEW

**Comment by alexdesiqueira** [comment]
Some questions arise from that, mainly in "Copies and views".
I think this section should be rewritten with other examples. Thoughts?
Thanks!

**Comment by rossbar** [comment]
I think that the Quickstart Tutorial is due for an overhaul soon (cf. NEP 44) but even so, I think the changes related to `.shape` are a good idea. I don't think modifying the `.shape` attribute is something that should be encouraged, even though it is done here for illustrative purposes.

**Comment by alexdesiqueira** [comment]
>  I don't think directly modifying the `.shape` attribute is something that should be encouraged, even though it is done here for illustrative purposes.

Even more in a "quickstart guide", which I think could be seen by thousands of beginners.

**Comment by charris** [comment]
>I don't think directly modifying the .shape attribute is something that should be encouraged, even though it is done here for illustrative purposes.

At some point we considered making that an error, but decided not to for backwards compatibility. There may even have been case where it was needed but I cannot recall what that was :)

**Comment by rgommers** [comment]
> I think that the Quickstart Tutorial is due for an overhaul soon (cf. NEP 44)

That's not high on our list, feel free to make any improvements you want there. NEP 44 is first of all about restructuring, second about creating space for better tutorials, how-to's, etc. Rewriting existing outdated content comes only after that, and quickstart isn't specifically on the todo list.

File: doc/source/user/quickstart.rst [hunk 1]
```diff
@@ -1441,7 +1437,7 @@ To change the dimensions of an array, you can omit one of the sizes
 which will then be deduced automatically::
 
     >>> a = np.arange(30)
-    >>> a.shape = 2,-1,3  # -1 means "whatever is needed"
+    >>> a = a.reshape((2, -1, 3))  # -1 means "whatever is needed"
```
Comment (mattip) [inline doc/source/user/quickstart.rst]: maybe change the returned value's name to make it clear this is a new object: `b = a.reshape(...`

**Comment by mattip** [comment]
sorry to hijack this PR with CI issues but ...
@tylerjereddy any idea what is up with the azure macOS image? When trying to install gfortran it complains `ln: /usr/local/bin/gfortran: File exists`

**Comment by mattip** [comment]
It would be nice to verify this passes the refguide-check, but that is blocked by the macOS build failing to install gfortran

**Comment by mattip** [comment]
close/reopen to see if the recently-merged change to macOS gets pulled in to CI builds

**Comment by mattip** [comment]
Thanks @alexdesiqueira

**Pull request merged**